src/cmd/flash.js

```diff
@@ -24,16 +27,42 @@ module.exports = class FlashCommand {
 
 		let result;
 		if (usb){
-			result = this.flashDfu({ binary, factory, force });
+			if (files.length > 0) {
+				// If both a device and a binary were provided, the binary will come in as the
+				// first element in the files array
+				binary = files[0];
+			} else {
+				// Otherwise, no device has been provided so unset the inaccurate value
+				device = undefined;
+			}
+
+			// Lookup the Device ID based on the provided Device Name
+			if (device && !isDeviceId(device)) {
+				const api = new ParticleApi(settings.apiUrl, { accessToken: settings.access_token }).api;
```

Comment:
at some point i think we're going to have to insist of device IDs across the board as this type of thing leads to confusion - e.g. "why am i getting 4xx errors when flashing locally?!" etc